src/compiler/types.ts

```diff
@@ -1,3 +1,6 @@
+// TODO: remove once tslint allows tsconfig.json to include lib "esnext.bigint"
+declare const BigInt: ((value: number | string) => number) | undefined;

```

Comment:
not sure i understand why we need this in the compiler.

Reply:
I would say just use the internal representation always. either ways, this should be defined close to where it is used, and should be marked as `/* @internal */`

Reply:
I may not have explained clearly how `bigint` literals are represented internally. They are stored as `{negative: boolean, base10Value: string}`, where `negative` is true iff the value is less than 0 and `base10Value` is the value of the literal as a decimal string. For example, `0xFFn` would be represented as `{negative: false, base10Value: "255"}`. I chose to store the value in decimal so that type strings) can easily be generated (e.g. `const a = 0xFFn` would declare `a` as type `255n`. It also unifies literals expressed in different bases, so `0b10000n`,`0x10n`, and `16n` are all considered the same type. This representation is used regardless of whether the `BigInt` global is defined.

`BigInt` is only used (if it is available) to convert literal strings in different bases to the decimal string. So `0xFFn` would be parsed using `"" + BigInt("0xFF")`. If `BigInt` is not available, the code falls back to computing a `Uint16Array` with the bits of the literal's value, and then computing the decimal digits of this value by repeatedly dividing by 10 and storing the remainder.

I will move the declaration to where it is used and mark it as `internal`.